modules/serverpod_auth2/serverpod_auth_profile/serverpod_auth_profile_server/lib/src/business/user_profiles.dart

```diff
@@ -0,0 +1,218 @@
+  // static Future<UserInfo?> createUser(
+  //   Session session,
+  //   UserInfo userInfo, [
+  //   String? authMethod,
+  //   UserInfoCreationCallback? onUserWillBeCreatedOverride,
+  //   UserInfoUpdateCallback? onUserCreatedOverride,
+  // ]) async {
+  //   bool approved = switch (onUserWillBeCreatedOverride) {
+  //     null => await AuthConfig.current.onUserWillBeCreated?.call(
+  //           session,
+  //           userInfo,
+  //           authMethod,
+  //         ) ??
+  //         true,
+  //     _ => await onUserWillBeCreatedOverride.call(session, userInfo, authMethod)
+  //   };
+  //   if (!approved) return null;
+
+  //   userInfo = await UserInfo.db.insertRow(session, userInfo);
+
+  //   if (onUserCreatedOverride == null) {
+  //     await AuthConfig.current.onUserCreated?.call(session, userInfo);
+  //   } else {
+  //     await onUserCreatedOverride(session, userInfo);
+  //   }
+
+  //   return userInfo;
+  // }
+
+  // /// Finds a user by its email address. Returns null if no user is found.
+  // static Future<UserInfo?> findUserByEmail(
+  //   Session session,
+  //   String email,
+  // ) async {
+  //   return await UserInfo.db.findFirstRow(
+  //     session,
+  //     where: (t) => t.email.equals(email),
+  //   );
+  // }
+
+  // /// Finds a user by its sign in identifier. For Google sign ins, this is the
+  // /// email address. For Apple sign ins, this is a unique identifying string.
+  // /// Returns null if no user is found.
+  // static Future<UserInfo?> findUserByIdentifier(
+  //   Session session,
+  //   String identifier,
+  // ) async {
+  //   return await UserInfo.db.findFirstRow(
+  //     session,
+  //     where: (t) => t.userIdentifier.equals(identifier),
+  //   );
+  // }
+
+  static String userProfileCacheKey(
+    /// ID of then [AuthUser]
+    UuidValue userId,
+  ) {
+    return 'serverpod_auth_profile_$userId';
+  }
+
+  /// Find a user by its id. Returns null if no user is found. By default the
+  /// result is cached locally on the server. You can configure the cache
+  /// lifetime in [UserProfileConfig], or disable it on a call to call basis by
+  /// setting [useCache] to false.
+  static Future<UserProfile?> findUserByUserId(
+    Session session,
+    UuidValue userId, {
+    bool useCache = true,
+  }) async {
+    var cacheKey = userProfileCacheKey(userId);
+    UserProfile? userInfo;
+
+    if (useCache) {
+      userInfo = await session.caches.local.get<UserProfile>(cacheKey);
+      if (userInfo != null) return userInfo;
+    }
+
+    userInfo = await UserProfile.db.findFirstRow(
+      session,
+      where: (t) => t.userId.equals(userId),
+    );
+
+    if (useCache && userInfo != null) {
+      await session.caches.local.put(
+        cacheKey,
+        userInfo,
+        lifetime: UserProfileConfig.current.userInfoCacheLifetime,
+      );
+    }
+
+    return userInfo;
+  }
+
+  /// Updates a users name.
+  static Future<UserProfile> changeUserName(
```

Comment:
Other than the `serverpod_auth` methods, these would now throw if the user is not found.

As I don't think there is any reasonable case where the user account "just disappeared" while the request was being worked on, and it seems to easy to e.g. confuse the "user profile ID" and "user ID", so we really ought to give some feedback if the update went nowhere.